src/main/webapp/app/modules/user-profile/user-profile-role.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps } from 'react-router-dom';
+import '../../shared/components/listing-card/listing-card.scss';
+
+import { IRootState } from 'app/shared/reducers';
+import { Row, Col } from 'reactstrap';
+import { CCRoleType } from 'app/shared/model/user-cc-role-info.model';
+import { getCurrentUserCCRolesProfile } from './user-profile.reducer';
+
+export interface IUserProfileRoleProps extends StateProps, DispatchProps, RouteComponentProps<{}> {}
+
+class UserProfileRole extends React.Component<IUserProfileRoleProps, {}> {
+  constructor(props: IUserProfileRoleProps) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.props.getCurrentUserCCRolesProfile();
+  }
+
+  render() {
+    const { userCCRoleInfo } = this.props;
+    return (
+      <>
+        {userCCRoleInfo.length > 0 && (
+          <div>
+            {userCCRoleInfo.map((roleInfo, index) => (
+              <CCRoleCard
+                key={`role-info-${index}`}
+                roleName={roleInfo.role}
+                roleType={roleInfo.type}
+                yearSession={roleInfo.yearSession}
+                eventName={roleInfo.eventName}
+              />
+            ))}
+          </div>
+        )}
+      </>
+    );
+  }
+}
+
+const mapStateToProps = ({ user }: IRootState) => ({
+  userCCRoleInfo: user.userCCRolesInfo,
+});
+
+const mapDispatchToProps = {
+  getCurrentUserCCRolesProfile,
+};
+
+type StateProps = ReturnType<typeof mapStateToProps>;
+type DispatchProps = typeof mapDispatchToProps;
+
+export default connect(mapStateToProps, mapDispatchToProps)(UserProfileRole);
+
+interface ICCRoleCardProps {
+  key: string;
+  roleType?: CCRoleType;
+  roleName?: string;
+  yearSession?: string;
+  eventName?: string;
```

Comment:
Why use optional property?

Reply:
I can change all to not optional
Except for eventName because user role might not belong to event

Reply:
resolved in 64f2e1f0dc946515016105ce79b1f131146ef918